### PR TITLE
Improve ETLv2 query debug

### PR DESCRIPTION
**Pull request by smgallo**
Improve ETLv2 query debuging

## Description

Augment Loggable::logAndThrowException() to accept a list of options and customize messages based on these options. This allowed us to merge Loggable::logAndThrowSqlException() into Loggable::logAndThrowException() and improve the display of SQL in debug mode.

Also display the DataEndpoint associated with every query now and fix phpcs style issues.

**Note that commits 216aa24, 8bfbcac, 729b1da were from a local merge and have already been merged in #45.**

## Motivation and Context

ETL is getting more complex pulling data from multiple databases, schemas, and files. It is important when debugging to know the queries being generated and the databases and schemas (DataEndpoints) that they are being run against.

## Tests performed

Ran the following ETLv2 pipelines: xdcdb-jobs (with and without aggregation binning), test-suite, resource-allocations. These test the following actions:
- Database Ingestor
- Update ingestor
- Jobs Aggregator
- Simple Aggregator
- Execute SQL
- MySQL, Postgres, JSON data endpoints

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project as found in the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: classes/ETL/Ingestor/RestIngestor.php [hunk 1]
```diff
@@ -916,19 +930,20 @@ private function verifyTransformDirective($key, \stdClass $directive)
...
-            if ( false === preg_match($directive->format, "test") ) {
-                $msg = "Invalid regex format '{$directive->format}' for key '$key'";
+            case 'datetime':
+                break;
+            case 'sprintf':
+                break;
+            case 'regex':
+                if ( false === preg_match($directive->format, "test") ) {
+                    $msg = "Invalid regex format '{$directive->format}' for key '$key'";
+                    $this->logAndThrowException($msg);
```
Comment (plessbd) [inline classes/ETL/Ingestor/RestIngestor.php]: do we need to set the error message to a variable or should we just pass it into the function call?

File: classes/ETL/Ingestor/RestIngestor.php [hunk 2]
```diff
@@ -916,19 +930,20 @@ private function verifyTransformDirective($key, \stdClass $directive)
...
+                }
+                break;
+            default:
+                $msg = "Unsupported transform type '{$directive->type}' for key '$key'";
```
Comment (plessbd) [inline classes/ETL/Ingestor/RestIngestor.php]: do we need to set the error message to a variable or should we just pass it into the function call?

File: classes/ETL/Ingestor/RestIngestor.php [hunk 3]
```diff
@@ -955,16 +970,16 @@ private function verifyVerifyDirective($key, \stdClass $directive)
         }
 
         switch ( $directive->type ) {
-        case 'regex':
-            if ( false === preg_match($directive->format, "test") ) {
-                $msg = "Invalid regex format '{$directive->format}' for key '$key'";
+            case 'regex':
+                if ( false === preg_match($directive->format, "test") ) {
+                    $msg = "Invalid regex format '{$directive->format}' for key '$key'";
```
Comment (plessbd) [inline classes/ETL/Ingestor/RestIngestor.php]: do we need to set the error message to a variable or should we just pass it into the function call?

File: classes/ETL/Ingestor/RestIngestor.php [hunk 4]
```diff
@@ -955,16 +970,16 @@ private function verifyVerifyDirective($key, \stdClass $directive)
...
+                    $this->logAndThrowException($msg);
+                }
+                break;
+            default:
+                $msg = "Unsupported transform type '{$directive->type}' for key '$key'";
```
Comment (plessbd) [inline classes/ETL/Ingestor/RestIngestor.php]: do we need to set the error message to a variable or should we just pass it into the function call?

File: classes/ETL/Ingestor/RestIngestor.php [hunk 5]
```diff
@@ -1018,24 +1033,24 @@ private function applyDirectives($value, \stdClass $directives)
...
-        default:
-            break;
+            case 'datetime':
+                $value = date($directive->format, strtotime($value));
+                break;
+            case 'regex':
+                $matches = null;
+                $matched = preg_match($directive->format, $value, $matches);
+                if ( false === $matched ) {
+                    $msg = "Error transforming regex '{$directive->format}'";
```
Comment (plessbd) [inline classes/ETL/Ingestor/RestIngestor.php]: do we need to set the error message to a variable or should we just pass it into the function call?

File: classes/ETL/Ingestor/RestIngestor.php [hunk 6]
```diff
@@ -1057,19 +1072,18 @@ private function applyTransformDirective($value, \stdClass $directive)
...
-                $msg = "Failed {$directive->type} ({$directive->format}) verification for '$value'";
-                $this->logAndThrowException($msg);
-            }
-            break;
-        default:
-            break;
+            case 'regex':
+                $matched = preg_match($directive->format, $value);
+                if ( 0 === $matched ) {
+                    $msg = "Failed {$directive->type} ({$directive->format}) verification for '$value'";
```
Comment (plessbd) [inline classes/ETL/Ingestor/RestIngestor.php]: do we need to set the error message to a variable or should we just pass it into the function call?

File: classes/ETL/Ingestor/UpdateIngestor.php [hunk 7]
```diff
@@ -211,39 +210,62 @@ public function execute(EtlOverseerOptions $etlOptions)
...
-            implode(", ", array_map(function($s) { return "$s = ?"; }, $this->parsedDefinitionFile->update->set)) .
-            " WHERE " .
-            implode(" AND ", array_map(function($w) { return "$w = ?"; }, $this->parsedDefinitionFile->update->where));
-        
-        $this->logger->debug("Update query\n$sql"); 
-       
+        $sql = "UPDATE " . $this->etlDestinationTable->getFullName() . " SET "
+            . implode(
+                ", ",
+                array_map(
```
Comment (plessbd) [inline classes/ETL/Ingestor/UpdateIngestor.php]: could/should we turn these two into a function?  pass the connector (', ' or ' AND ' ) and the values?

**Comment by smgallo** [comment]
Addressed @plessbd review comments. The change to UpdateIngestor.php is out of the scope of this pull request but there is a plan to deal with that by moving query construction tools into their own utility class.

**Pull request merged**